app/controllers/application_controller.rb

```diff
@@ -5,6 +5,138 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
 
+  get '/' do 
+    erb :index
+  end
+
+  get '/signup' do 
+    if session[:user_id]
```

Comment:
You can use your `logged_in?` helper method here, which is preferable to referring to the session object directly.